### PR TITLE
fix: spdk replica lists its own lvols so that everyone won't be affected by others

### PR DESCRIPTION
**Pull request by shuo-wu**
#### Which issue(s) this PR fixes:

longhorn/longhorn#9121

## Summary by CodeRabbit

- **New Features**
	- Introduced filtering for logical volume operations, enhancing precision during synchronization and deletion.
	- Added a new method for retrieving logical volumes with customizable filters.

- **Bug Fixes**
	- Improved error handling and logging related to the new filtering mechanism.

- **Chores**
	- Updated Go version and dependencies to ensure compatibility and access to new features.

### CODE REVIEW

**Comment by DamiaSan** [comment]
LGTM

**Review by derekbit — APPROVED** [review]
LGMT

File: pkg/spdk/replica.go [hunk 1]
```diff
@@ -205,7 +205,14 @@
...
 	// This lvol and nvmf subsystem fetch should be protected by replica lock, in case of snapshot operations happened during the sync-up.
-	bdevLvolMap, err := GetBdevLvolMap(spdkClient)
+	replicaLvolFilter := func(bdev *spdktypes.BdevInfo) bool {
+		var lvolName string
+		if len(bdev.Aliases) == 1 {
+			lvolName = spdktypes.GetLvolNameFromAlias(bdev.Aliases[0])
+		}
+		return IsReplicaLvol(r.Name, lvolName) || (r.ActiveChain[0] != nil && r.ActiveChain[0].Name == lvolName)
+	}
+	bdevLvolMap, err := GetBdevLvolMapWithFilter(spdkClient, replicaLvolFilter)
```
[the remark this answers is not in the document]
Comment (shuo-wu) [reply pkg/spdk/replica.go:215]: @derekbit Do I need to create a separate PR for this fair suggestion

[the remark this answers is not in the document]
Comment (derekbit) [reply pkg/spdk/replica.go:215]: > @derekbit Do I need to create a separate PR for this fair suggestion

Sure. Feel free to create it. Thank you.

**Pull request merged**